app/components/eto/overview/EtoOverviewThumbnail/Whitelist/Whitelist.tsx

```
@@ -0,0 +1,80 @@
+import * as React from "react";
+import { FormattedMessage } from "react-intl-phraseapp";
+import { setDisplayName, withProps } from "recompose";
+import { compose } from "redux";
+
+import { actions } from "../../../../../modules/actions";
+import { selectBookbuildingStats } from "../../../../../modules/bookbuilding-flow/selectors";
+import { TEtoWithCompanyAndContract } from "../../../../../modules/public-etos/types";
+import { appConnect } from "../../../../../store";
+import { onEnterAction } from "../../../../../utils/OnEnterAction";
+import { CounterWidget } from "../CounterWidget";
+import { WhitelistStatus } from "./WhitelistStatus";
+
+import * as styles from "../EtoStatusManager.module.scss";
+
+export interface IExternalProps {
+  eto: TEtoWithCompanyAndContract;
+  nextStateStartDate?: Date;
+}
+
+interface IStateProps {
+  pledgedAmount: number | null;
+  investorsCount: number;
+}
+
+interface IWithProps {
+  isInvestorsLimitReached: boolean;
+}
+
+type IProps = IWithProps & IExternalProps & IStateProps;
+
+const WhitelistLayout: React.FunctionComponent<IProps> = ({
+  eto,
+  pledgedAmount,
+  investorsCount,
+  isInvestorsLimitReached,
+  nextStateStartDate,
+}) => {
+  if (eto.isBookbuilding && !isInvestorsLimitReached) {
+    return (
+      <>
+        <WhitelistStatus
+          pledgedAmount={pledgedAmount}
+          investorsCount={investorsCount}
+          investorsLimit={eto.maxPledges}
+        />
+
+        <p className={styles.info}>
+          <FormattedMessage id="eto-overview-thumbnail.whitelist.is-open" />
+        </p>
+      </>
+    );
+  }
+
+  return <CounterWidget endDate={nextStateStartDate!} />;
```

maybe we should have a fallback to if `nextStateStartDate` is not there somehow `nextStateStartDate!` looks tricky

---

ok, let's just throw an exception in `EtoStatusManager` when timedState is not defined. As normally this should never happen.

---

:+1: